homeassistant/components/climate/mysensors.py

```diff
@@ -0,0 +1,216 @@
+"""
+mysensors platform that offers a Climate(MySensors-HVAC) component.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/climate.mysensors
+
+Curently supported operations:
+
+set_temperature()    --> V_HVAC_SETPOINT_COOL, V_HVAC_SETPOINT_HEAT
+set_operation_mode() --> V_HVAC_FLOW_STATE
+set_fan_mode()       --> V_HVAC_SPEED
+
+Pending:
+humidity, away_mode, aux_heat, swing_mode
+
+home-assistant/components/climate/mysensors.py
+"""
+import logging
+
+from homeassistant.components import mysensors
+from homeassistant.components.climate import (
+    STATE_COOL, STATE_HEAT, STATE_OFF, STATE_AUTO, ClimateDevice,
+    ATTR_TARGET_TEMP_HIGH, ATTR_TARGET_TEMP_LOW)
+from homeassistant.const import TEMP_CELSIUS, TEMP_FAHRENHEIT, ATTR_TEMPERATURE
+
+_LOGGER = logging.getLogger(__name__)
+
+DICT_HA_TO_MYS = {STATE_COOL: "CoolOn", STATE_HEAT: "HeatOn",
+                  STATE_AUTO: "AutoChangeOver", STATE_OFF: "Off"}
+DICT_MYS_TO_HA = {"CoolOn": STATE_COOL, "HeatOn": STATE_HEAT,
+                  "AutoChangeOver": STATE_AUTO, "Off": STATE_OFF}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the mysensors climate."""
+    if discovery_info is None:
+        return
+    for gateway in mysensors.GATEWAYS.values():
+        pres = gateway.const.Presentation
+        set_req = gateway.const.SetReq
+        map_sv_types = {
+            pres.S_HVAC: [set_req.V_HVAC_FLOW_STATE],
+        }
+        devices = {}
+        gateway.platform_callbacks.append(mysensors.pf_callback_factory(
+            map_sv_types, devices, add_devices, MySensorsHVAC))
+
+
+# pylint: disable=too-many-arguments, too-many-public-methods
+class MySensorsHVAC(mysensors.MySensorsDeviceEntity, ClimateDevice):
+    """Representation of a MySensorsHVAC hvac."""
+
+    def __init__(self, *args):
```

Comment:
OK... actually i was also confused as by keeping as well as deleting the code it was still working... had taken the safe bet of just keeping it there ;)